lib/srv/server/installer/autodiscover.go

```diff
@@ -325,14 +333,38 @@ func (ani *AutoDiscoverNodeInstaller) configureTeleportNode(ctx context.Context,
 		}
 
 		if hashExistingFile == hashNewFile {
+			if err := os.WriteFile(discoverNoticeFile, []byte(discoverNotice), 0o644); err != nil {
+				return trace.Wrap(err)
+			}
+
 			return trace.AlreadyExists("teleport.yaml is up to date")
 		}
+
+		// If a previous /etc/teleport.yaml configuration file exists and is different from the target one, it might be because one of the following reasons:
+		// - discover installation params (eg token name) were changed
+		// - `$ teleport node configure` command produces a different output
+		// - teleport was manually installed / configured
+		//
+		// For the first two scenarios, it's fine, and even desired in most cases, to restart teleport with the new configuration.
+		//
+		// However, for the last scenario (teleport was manually installed), this flow must not replace the currently running teleport service configuration.
+		// To prevent this, this flow checks for the existence of the discover notice file, and only allows replacement if it does exist.
+		if _, err := os.Stat(discoverNoticeFile); err != nil {
+			ani.Logger.InfoContext(ctx, "Refusing to replace the existing teleport configuration. For the script to replace the existing configuration remove the Teleport configuration.).",
```

Comment:
```suggestion
			ani.Logger.InfoContext(ctx, "Refusing to replace the existing teleport configuration. For the script to replace the existing configuration remove the Teleport configuration.",
```

Reply:
I went ahead and just committed the typo fix so I could re-add this to merge queue